CONTRIBUTING.md

```diff
@@ -26,26 +26,156 @@
   * Security  - non-upgrade-impacting
   * Bug fixes - non-upgrade-impacting
 
-## Working in the repository
+# Issue workflow
+
+### Definitions
+#### Issues
+Issues are work items, they may be bugs or enhancements. They are managed using [Waffle](https://waffle.io/rcbops/u-suk-dev?source=rcbops%2Frpc-openstack) and stored in [GitHub](https://github.com/rcbops/rpc-openstack/issues). Issues that are logged in other places, for example on Launchpad for OSA, are not within scope unless they have been reported on this project by the creation of a new issue to track the upstream item.
+
+#### Waffle board
+The [Waffle board](https://waffle.io/rcbops/u-suk-dev?source=rcbops%2Frpc-openstack) is a Kanban board consisting of a series of columns used to manage the progress of issues. This board is the only place to view or update the current work commitments.
+
+#### WIP (Work In Progress) limits
+Each column on the board has a WIP limit that is used to control the amount of new work to which the engineering team commits. The WIP limits will be listed in the column titles on the Waffle board. No column can receive new work if it violates the WIP limit unless there is an explicit exception made for it as defined in the processes below.
+
+#### Stakeholders
+* reporter - the individual that created the issue.
+* owner(s) - the individual(s) assigned to manage the issue through to its closure.
+* reviewer(s) - the individual(s) who review a pull request and provide feedback.
+
+#### Labels
+Labels are used to categorise issues to aid with tracking and prioritisation.
+
+|Type                     |Label name              |Description                                                                                                                                                   |Examples                                                                                      |
+|-------------------------|------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------|
+|Issue type               |type-bug                |Defects affecting the RPCO project.                                                                                                                           |                                                                                              |
```

Comment:
@prometheanfire What are you concerns regarding small/large bugs? 

Once we adopt a WIP limit approach and start to use the metrics data, the size of the bug doesn't really matter. My expectation is that when committing to a bug, that we look at the work required to fix the bug and ask whether we can complete it within our cycle time. If our cycle time for most bugs is about 5 days and if the bug can be completed within those 5 days, great, size doesn't matter. However, if the bug appears to be larger than the cycle time of 5 days, than size does matter. If we feel a bug will take longer to fix than our historical average, then we can look into breaking the bug out or even doing a mini-spike to determine if there's a way that we can do it within the cycle time.

The other scenario would be that if we commit to a bug that we thought would take only 5 days but it's on day 6, then obviously we need to look at it as a team because the bug is an outlier that needs to be resolved. So that's a swarming or pair programming opportunity.